addOns/client/src/main/java/org/zaproxy/addon/client/internal/ClientMapWriter.java

```diff
@@ -0,0 +1,202 @@
+                                .disable(Feature.WRITE_DOC_START_MARKER));
+        // For some reason the disable start marker doesn't seem to work
+        String output = mapper.writeValueAsString(value).replace("--- ", "");
+        fw.write(output);
+        return false;
+    }
+
+    private static void outputNode(Writer fw, ClientNode node, int level) throws IOException {
+        if (node.isStorage()) {
+            // Skip storage nodes in the tree
+            // Those details are represented as components of their source
+            return;
+        }
+        // We could create a set of data structures and use jackson, but the format is very
+        // simple and this is much more memory efficient - it still uses jackson for value
+        // output
+        String indent = " ".repeat(level * 2);
+
+        outputKV(
+                fw,
+                indent,
+                true,
+                NODE_KEY,
+                level == 0 ? ROOT_NODE_NAME : node.getUserObject().getName());
+
+        if (node.getUserObject().isStorage()) {
+            outputKV(fw, indent, false, STORAGE_KEY, node.getUserObject().isStorage());
+        }
+        if (!node.getUserObject().isVisited()) {
+            outputKV(fw, indent, false, VISITED_KEY, node.getUserObject().isVisited());
+        }
+        if (node.getUserObject().getComponents() != null
+                && !node.getUserObject().getComponents().isEmpty()) {
+            for (ClientSideComponent component : node.getUserObject().getComponents()) {
+                if (component.getTypeForDisplay().equalsIgnoreCase(ClientSideComponent.REDIRECT)) {
+                    outputKV(
+                            fw,
+                            indent,
+                            false,
+                            ClientSideComponent.REDIRECT.toLowerCase(Locale.ROOT),
+                            component.getHref());
+                }
+                if (component
+                        .getTypeForDisplay()
+                        .equalsIgnoreCase(ClientSideComponent.CONTENT_LOADED)) {
+                    outputKV(fw, indent, false, "contentLoaded", true);
+                }
+            }
+            indent = outputComponents(fw, node.getUserObject().getComponents(), level, indent);
+        }
+
+        Stats.incCounter(PopupMenuExportClientMap.STATS_EXPORT_CLIENTMAP + ".node");
+
+        if (node.getChildCount() > 0) {
+            fw.write(indent);
+            fw.write("  ");
+            fw.write(CHILDREN_KEY);
+            fw.write(":");
+            fw.write('\n');
+            node.children()
+                    .asIterator()
+                    .forEachRemaining(
+                            c -> {
+                                try {
+                                    outputNode(fw, (ClientNode) c, level + 1);
+                                } catch (IOException e) {
+                                    LOGGER.error(e.getMessage(), e);
```

Comment:
It should rethrow.

Reply:
How do I re-throw inside a lambda?

Reply:
⭐

Reply:
The catch block in the extension needs to be updated accordingly.